packages/elements/src/internals/machines/sign-in/machines/verification.machine.ts

```diff
@@ -53,6 +54,38 @@ const SignInVerificationMachine = setup({
     determineStartingFactor: () => {
       throw new ClerkElementsRuntimeError('Action `determineStartingFactor` must be overridden');
     },
+    validateConfiguredStrategies: ({ context }) => {
+      const clerk = context.parent.getSnapshot().context.clerk;
+
+      if (clerk.__unstable__environment?.isProduction()) {
+        return;
+      }
+
+      if (
+        !clerk.client.signIn.supportedFirstFactors.every(factor =>
+          context.registeredStrategies.has(factor.strategy as unknown as SignInFactor),
+        )
+      ) {
+        console.warn(
+          `Clerk: Your instance is configured to support strategies: ${clerk.client.signIn.supportedFirstFactors
+            .map(f => f.strategy)
+            .join(', ')}, but the rendered strategies are: ${[...context.registeredStrategies]
+            .map(s => s)
+            .join(
+              ', ',
+            )}. Before deploying your app, make sure to render a <Strategy> component for each supported strategy. For more information, visit the documentation: <link>`,
+        );
+      }
+
+      if (
+        process.env.NODE_ENV === 'development' &&
+        !context.registeredStrategies.has(context.currentFactor?.strategy as unknown as SignInFactor)
+      ) {
+        throw new ClerkElementsRuntimeError(
+          `Your sign-in/up attempt is missing a ${context.currentFactor?.strategy} strategy. Make sure <Strategy name="${context.currentFactor?.strategy}"> is rendered in your flow. For more information, visit the documentation: <link>`,
```

Comment:
This machine is only for sign-in, so we can get rid of `/up`.

Reply:
https://github.com/clerk/javascript/pull/3165/commits/543291c6266368749103d410c40e038f30bb7017